gempython/tools/optohybrid_user_functions_uhal.py

```diff
@@ -696,12 +703,20 @@ def getScanResults(device, gtx, numpoints, debug=False):
 
     return results
 
-def getUltraScanResults(device, gtx, numpoints, debug=False):
+def getUltraScanResults(device, gtx, numpoints, debug=False, numtrigs=1000, isLatency=False):
     scanBase = "GEM_AMC.OH.OH%d.ScanController.ULTRA"%(gtx)
+    ohnL1A_0 = getLinkL1Acount(device,gtx)
+    ohnL1A = getLinkL1Acount(device,gtx)
     while (readRegister(device,"%s.MONITOR.STATUS"%(scanBase)) > 0):
         msg = "%s: Ultra scan still running (0x%x), not returning results"%(device,
                                                                             readRegister(device,"%s.MONITOR.STATUS"%(scanBase)))
         ohlogger.debug(colormsg(msg,logging.DEBUG))
+        if (isLatency):
+            if ((getLinkL1Acount(device,gtx)-ohnL1A) > numtrigs):
+                print "At link %s: %d/%d L1As processed, %d%% done" %(gtx, getLinkL1Acount(device,gtx)-ohnL1A_0, numpoints*numtrigs, (getLinkL1Acount(device,gtx)-ohnL1A_0)*100./(numpoints*numtrigs))
```

Comment:
I agree that the printing should be somewhat dynamic.
I don't think it's very important to see an update at a fixed (in time) interval, the important thing is to simply see that the scan is progressing.  Some combination of the number of triggers per point and some number of updates during each point works, as does printing out at regular completion percentages.
You could consider something like [this](https://github.com/jsturdy/vfatqc-python-scripts/blob/dacscans/dacScan1.py#L7)
* You could print out something simple like:
```
<DAC name> <value>: <progress bar> | <total/partial completion percentage>
```
with a new line for each DAC value e.g.,
```
Latency 34: ====================> | 100%
Latency 35: ====================> | 100%
Latency 36: ============>         |  60%
```
or an overwritten line (where the progress bar is always moving up, but the latency value can change, as well as the percentage number, such that percentage is total, rather than per point):
```
Latency 36: ======>               | 37%
```
* Or something a little more fancy like (but requiring a bit more trickery):
```
<total progress bar> | <total completion percentage>
<DAC name> <value>: <DAC progress bar> | <DAC completion percentage>
```
e.g.,
```
Scan      : ===>                  |  23%
Latency 34: ====================> | 100%
Latency 35: ====================> | 100%
Latency 36: ============>         |  60%
```